packages/components/src/utils/form/isParentFieldsetRequired.spec.ts

```diff
@@ -1,50 +1,57 @@
 import { isParentFieldsetRequired } from './isParentFieldsetRequired';
-import type { HTMLElementWithRequiredProp } from './isRequired';
-
-// TODO: these tests should verify that isParentOfKind() and isRequired() are called with correct parameters and return their combined result
-describe('without prefix', () => {
-  it('should return true if parent is required', () => {
-    const parent = document.createElement('p-fieldset-wrapper');
-    const child = document.createElement('div');
-    parent.appendChild(child);
-    parent['required'] = true;
-
-    expect(isParentFieldsetRequired(child)).toBe(true);
-  });
-
-  it('should return false if parent is not required', () => {
-    const parent = document.createElement('p-fieldset-wrapper');
-    const child = document.createElement('div');
-    parent.appendChild(child);
-
-    expect(isParentFieldsetRequired(child)).toBe(false);
-  });
-
-  it('should return false if parent is not p-fieldset-wrapper', () => {
-    const parent = document.createElement('div');
-    const child = document.createElement('div');
-    parent.appendChild(child);
-
-    expect(isParentFieldsetRequired(child)).toBe(false);
-  });
+import * as isParentOfKindUtils from '../dom/isParentOfKind';
+import * as isRequiredUtils from './isRequired';
+
+it('should call isParentOfKind with correct parameters', () => {
+  const spy = jest.spyOn(isParentOfKindUtils, 'isParentOfKind');
+  const child = document.createElement('div');
+
+  isParentFieldsetRequired(child);
+
+  expect(spy).toBeCalledWith(child, 'p-fieldset');
+  expect(spy).toBeCalledWith(child, 'p-fieldset-wrapper');
+  expect(spy).toBeCalledTimes(2);
+});
+
+it('should call isRequired with correct parameters', () => {
+  const spy = jest.spyOn(isRequiredUtils, 'isRequired');
+  const parent = document.createElement('p-fieldset-wrapper');
+  const child = document.createElement('div');
+  parent.appendChild(child);
+
+  isParentFieldsetRequired(child);
+
+  expect(spy).toBeCalledWith(parent);
 });
 
-describe('with prefix', () => {
-  it('should return true if parent is required', () => {
-    const parent = document.createElement('prefixed-p-fieldset-wrapper') as HTMLElementWithRequiredProp;
-    const child = document.createElement('div');
-    parent.appendChild(child);
-    parent.required = true;
+it('should return true if isRequired and isParentOfKind return true', () => {
```

Comment:
Perhaps having an `it.each` setup for the 4 scenarios would be easier to maintain an read.